main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"assignment-exec/image-builder/builder"
+	"assignment-exec/image-builder/environment"
+	"assignment-exec/image-builder/utilities"
+	"flag"
+	"fmt"
+	"github.com/pkg/errors"
+	"log"
+	"os"
+)
+
+var publishImage = flag.Bool("publishImage", false, "Push image to docker hub")
+var codeRunnerConfig = flag.String("codeRunnerConfig", "code-runner.yaml", "Code Runner configuration filename")
+var assignmentEnv = flag.String("assignmentEnv", "assignment-env.yaml", "Assignment Environment configuration filename")
+var dockerfileName = flag.String("dockerfile", "Dockerfile", "Dockerfile name")
+
+func main() {
+
+	flag.Parse()
+
+	authData := builder.GetAuthData()
+
+	err := generateCodeRunnerImage(authData)
+	if err != nil {
+		log.Fatalf("error while building code runner image: %v", err)
+	}
+
+	err = generateAssignmentEnvImage(authData)
+	if err != nil {
+		log.Fatalf("error while building assignment environment image: %v", err)
+	}
+
+}
+
+// Generate a dockerfile for code runner server, build its image and push it ot docker hub.
+func generateCodeRunnerImage(authData *builder.DockerAuthData) error {
+
+	// Unmarshal the yaml configuration file and generate a dockerfileName.
+	err, _, _ := utilities.WriteDockerfile(*codeRunnerConfig, *dockerfileName)
+	if err != nil {
+		return errors.Wrap(err, "error in writing dockerfile for code runner")
+	}
+
+	repositoryName := os.Getenv(environment.CodeRunnerRepository)
```

Comment:
What if the environment variables are not defined?

Reply:
Fixed. Used `os.LookupEnv(...)` and handled error